src/androidTest/java/com/nutomic/syncthingandroid/test/syncthing/AppConfigReceiverTest.java

```diff
@@ -0,0 +1,169 @@
+import android.preference.PreferenceManager;
+import android.test.AndroidTestCase;
+
+import com.nutomic.syncthingandroid.syncthing.AppConfigReceiver;
+import com.nutomic.syncthingandroid.syncthing.SyncthingService;
+import com.nutomic.syncthingandroid.test.MockContext;
+
+import junit.framework.Assert;
+
+/**
+ * Test the correct behaviour of the AppConfigReceiver
+ *
+ * Created by sqrt-1674 on 27.03.16.
+ */
+public class AppConfigReceiverTest extends AndroidTestCase {
+    private AppConfigReceiver mReceiver;
+    private MockContext mContext;
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        mReceiver = new AppConfigReceiver();
+        mContext = new MockContext(getContext());
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        PreferenceManager.getDefaultSharedPreferences(mContext).edit().clear().commit();
+        super.tearDown();
+    }
+
+
+    /**
+     * Test starting and stopping the Syncthing-Service if "always run in background" is enabled
+     * In this case starting the service is allowed
+     */
+    public void testStartSyncthingServiceBackground() {
+        PreferenceManager.getDefaultSharedPreferences(mContext)
+                .edit()
+                .putBoolean(SyncthingService.PREF_ALWAYS_RUN_IN_BACKGROUND, true)
+                .commit();
+
+        Intent intent = new Intent(new Intent(mContext, AppConfigReceiver.class));
+        intent.setAction(AppConfigReceiver.ACTION_START);
+
+        mReceiver.onReceive(mContext, intent);
+        assertEquals("Start SyncthingService Background", 1, mContext.getReceivedIntents().size());
+        assertEquals("Start SyncthingService Background", SyncthingService.class.getName(), mContext.getReceivedIntents().get(0).getComponent().getClassName());
+    }
+
+    /**
+     * Method #1:
+     * Test stopping the service if "alway run in background" is enabled.
+     * In this mode stopping the service is not allowed. In that case a notification is created.
+     * (This crashes in this test context which indicates that the test succeeded.)
+     */
+    public void testStopSyncthingServiceBackground() {
+        PreferenceManager.getDefaultSharedPreferences(mContext)
+                .edit()
+                .putBoolean(SyncthingService.PREF_ALWAYS_RUN_IN_BACKGROUND, true)
+                .commit();
+
+        Intent intent = new Intent(new Intent(mContext, AppConfigReceiver.class));
+        intent.setAction(AppConfigReceiver.ACTION_STOP);
+
+        // onReceive now tries to create a notification.
+        // This is expected to throw an exception as creating a notification is not supported in this Test-Context
+        // so an exception is an indicator of success in this case.
+        try {
+            mReceiver.onReceive(mContext, intent);
+            Assert.fail("Stop SyncthingService Background (1) - didn't fail to create the notification");
+
+        } catch (NullPointerException e) {
```

Comment:
I don't understand either why it crashes, seeing as any calls are passed through to the normal context.

I don't really know what we should do here. But getContentResolver() is called directly before the NPE occurs. That means we can check that call for the same result, and without the side effect of hiding all other NPEs.